feat(*): Update to Helm v3

hickeyma opened the pull request:
Helm v3 is now the current release of Helm. It replaces Helm v2 and there are changes which differs it from Helm v2.

This PR updates the project to be compliant with Helm v3. Helm v2 will be maintained in the following branch: https://github.com/IBM/helm101/tree/helm-v2

Supersedes #27

hickeyma committed:
Update to Helm v3

Helm v3 is now the current release of Helm. It replaces Helm v2
and there are changes which differs it fropm Helm v2.

hickeyma commented:
@lee-zhg This is the PR I mentioned to supersede #27 . I would appreciate if you could review this and provide your feedback.

spzala commented on README.md:
```diff
@@ -2,4 +2,20 @@
+
+## Helm Status
+
+[Helm v3 was released](https://helm.sh/blog/helm-3-released/) in November 2020. The interface is quite similar but there
```
November 2019 ? :)

spzala commented on README.md:
```diff
@@ -2,4 +2,20 @@
+
+The default branch of this repo is `master` and it uses the current major release of [Helm](https://helm.sh/) which is v3. If you
+are using Helm v2 then go to the [helm-v2](https://github.com/IBM/helm101/tree/helm-v2) branch. Please look below for more
+details on the different Helm releases.
```
I don't think "Please look below for more
details on the different Helm releases." is needed OR suggest to re-write "Please refer to the Helem Status section below for more
details on the different Helm releases."

hickeyma replied:
Sounds good. Updated.

spzala commented on README.md:
```diff
@@ -2,4 +2,20 @@
+The [Helm 2 Support Plan](https://helm.sh/blog/2019-10-22-helm-2150-released/#helm-2-support-plan) documents a 1 year "maintenance
+mode" for Helm v2. It states the following:
+- 6 month bug fixes until May 13 2020
+- 6 month security fixes until November 13 2020
```
Is it 6 month or 1 year as it stats until Nov 13?

hickeyma replied:
6 months bug fix + 6 months security fix == 1 year maintenance

spzala commented on tutorial/Lab0/README.md:
```diff
@@ -1,26 +1,20 @@
 # Installing the Helm Client (helm)
 
-1. Download the [latest release Helm](https://github.com/helm/helm/releases) for your environment, which in this case is `Linux amd64`.
+1. Download the [latest release of Helm v3](https://github.com/helm/helm/releases) for your environment, which in this case is `Linux amd64`.
```
Suggest to use "latest release of Helm" (take out v3)

hickeyma replied:
Unfortunately at the moment, the releases in GitHub will show in order when a releases is published. This mean that you could have a v2 releases showing before a v3 release. This is why I am specifying v3 here.

spzala commented on tutorial/Lab4/README.md:
```diff
@@ -6,7 +6,7 @@ A means of sharing is a chart repository, which is a location where packaged cha
 # Using charts from a public repository
 
-Helm charts can be available on a remote repository or in a local environment/repository. The remote repositories can be public like [Helm Charts](https://github.com/helm/charts) or [IBM Helm Charts](https://github.com/IBM/charts), or hosted repositories like on Google Cloud Storage or GitHub. Refer to [Helm Chart Repository Guide](https://v2.helm.sh/docs/developing_charts/#the-chart-repository-g ...
+Helm charts can be available on a remote repository or in a local environment/repository. The remote repositories can be public like [Bitnami Charts](https://github.com/bitnami/charts) or [IBM Helm Charts](https://github.com/IBM/charts), or hosted repositories like on Google Cloud Storage or GitHub. Refer to [Helm Chart Repository Guide](https://helm.sh/docs/topics/chart_repository/) for more det ...
```
Why change /helm/charts to /bitnami/charts?

hickeyma replied:
This is because the helm chart repository is going out of support in parallel with Helm v2 support.    Charts will ne need top be hosted by the chart maintainer. Details on this: https://github.com/helm/charts#status-of-the-project

spzala commented on tutorial/Lab4/README.md:
````diff
@@ -32,54 +30,45 @@ In this part of the lab, we show you how to install the `guestbook` chart from t
    
    ```"helmm101" has been added to your repositories```
````
can you please fix this too from helmm101 to helm101 :) thanks

hickeyma replied:
Good catch, fixed

spzala commented on tutorial/README.md:
```diff
@@ -13,31 +13,41 @@ These labs provide an insight on the advantages of using Helm over using Kuberne
   * Revision management
   * Repositories and chart sharing
 
+# Helm Status
```
Since it's same as main README, can we just provide a link something like "Refer to the README Helm Status section" so that we don't have any inconsistency in future accidentally?

hickeyma replied:
Good idea, updated.

spzala commented on tutorial/README.md:
```diff
@@ -13,31 +13,41 @@ These labs provide an insight on the advantages of using Helm over using Kuberne
 ![helm-architecture](images/helm-architecture.png)
 
-It has a client-server architecture with the client called `helm` and the server called `Tiller`. The client is a CLI which users interact with to perform different operations like install/upgrade/delete etc. The client interacts with Tiller and the chart repository. Tiller interacts with the Kubernetes API server. It renders Helm template files into Kubernetes manifest files which it uses to per ...
+It has a client architecture with the client called `helm` and a Go library which encapsulates the Helm logic so that it can be leveraged by different clients. The client is a CLI which users interact with to perform different operations like install/upgrade/delete etc. The client interacts with the Kubernetes API server. It renders Helm template files into Kubernetes manifest files which it uses ...
```
What are the different clients here?

hickeyma replied:
No supported clients, its just that the SDK allows users to plug in their own client. The only client supported by helm is `helm`.

spzala commented on tutorial/README.md:
```diff
@@ -13,31 +13,41 @@ These labs provide an insight on the advantages of using Helm over using Kuberne
 * Chart - It contains all of the resource definitions necessary to run an application, tool, or service inside of a Kubernetes cluster. A chart is basically a package of pre-configured Kubernetes resources.
 * Config - Contains configuration information that can be merged into a packaged chart to create a releasable object.
-* helm - Helm client. Communicates to Tiller through the Helm API - [HAPI](https://v2.helm.sh/docs/developers/#the-helm-api-hapi) which uses [gRPC](https://grpc.io/).
+* helm - Helm client. It renders charts into manifest fils. It interacts directly with the [Kubernetes API](https://kubernetes.io/docs/concepts/overview/kubernetes-api/) server to install, upgrade, query, and remove Kubernetes resources.
```
typo - fils vs files ?

hickeyma replied:
fixed

spzala reviewed (commented):
@hickeyma thanks, it looks good though I have few comments. PTAL.

hickeyma committed:
Update after review

Review comments:
- https://github.com/IBM/helm101/pull/31#pullrequestreview-369221720

hickeyma commented:
Thanks for the review and feedback @spzala.  Comments updated and ready for review again.

spzala commented:
@hickeyma yrw and thanks for addressing my comments. 
LGTM

hickeyma commented:
@spzala Thanks for the review. Merging following your LGTM?

spzala commented:
@hickeyma yrw, yes sounds good!
